AGPs/AGP-anteater1.md

```diff
@@ -0,0 +1,31 @@
+---
+AGP: N/A
+Title: Oppose any change to Ethereum's mining algorithm before Ethereum 2.0
+Author: anteater (@anteater0x)
+Status: Stage III
+Track: Proclamation
+Created: 2019-09-22
+---
+
+# AGP-X: Oppose any change to Ethereum's mining algorithm before Ethereum 2.0
+
+## Proclamation text
+
+ANT holders who vote "yes" on this proposal oppose Ethereum changing its proof-of-work mining algorithm before the switch to proof-of-stake as part of the Ethereum 2.0 roll-out, unless such change is intended as an "emergency fix" to a fatal flaw in the current mining algorithm. A "fatal flaw" is defined here as a flaw that breaks the expected functionality of Ethereum (e.g. if blocks are consistently mined too quickly or too slowly, or if blocks could be mined by some miners faster than others despite having equal hashpower) or causes Ethereum to completely stop working (i.e. stop producing blocks and confirming valid transactions).
```

Comment:
What do you mean by Ethereum 2.0 roll-out? Can you clarify whether this refers to the launch of the beacon chain (phase 0) or when Ethereum 2.0 has at least feature parity with Eth 1

Reply:
> What do you mean by Ethereum 2.0 roll-out? 

The process of implementing and activating the upgrades colloquially referred to as "[Ethereum 2.0](https://medium.com/chainsafe-systems/ethereum-2-0-a-complete-guide-d46d8ac914ce)".

> Can you clarify whether this refers to the launch of the beacon chain (phase 0) or when Ethereum 2.0 has at least feature parity with Eth 1

Whenever Ethereum implements and switches to proof-of-stake. This proposal opposes any change to Ethereum's proof-of-work mining algorithm at any time between now (if/ when the proposal is approved) and the time that Ethereum switches to proof-of-stake, except in case of emergency as described. The only mining algorithm change this proposal finds acceptable is a change from proof-of-work to proof-of-stake as planned for Ethereum 2.0.

This seems clear in the proposal language to me, but do you think there is a way this could be more clearly expressed?